JS/Lesson14/sinan/script.js

```diff
@@ -4,90 +4,119 @@
 // 10. Use the `replace` method to replace "can help" with "help" in the string stringExercise10 and print the result to the console.
 const stringExercise10 = "Software design patterns can help simplify complex can help systems.";
-console.log("Exercise10: ", stringExercise10.replace("can help", "help"));
+console.log('Exercises 10:',stringExercise10.replace("can help","help"));
+console.log('Exercises 10.1:',stringExercise10.replace(/ /g,"-"));
+
 
 // What's the best approach to find and replace all occurences?
 
 console.log('------------------------');
 // 11. Use the `startsWith` method to check if strings below starts with a price, and print the result to the console.
 const string1Exercise11 = "$9.99 - Sale! 50% off all items in store";
 const string2Exercise11 = "Bundle deal: Save $50 when you buy two products together";
-console.log("Exercise11: ", string1Exercise11.startsWith("$"));
-console.log("Exercise11: ", string2Exercise11.startsWith("$"));
+console.log('Exercises 11.1: ', string1Exercise11.startsWith("$"));
+console.log('Exercises 11.2: ', string2Exercise11.startsWith("$"));
+
 
 console.log('------------------------');
 // 12. Use the `endsWith` method to check if a string ends with a "!" (exclamation mark), and print the result to the console.
 const string1Exercise12 = "$9.99 - Sale! 50% off all items in store";
 const string2Exercise12 = "Bundle deal: Save $50 when you buy two products together!";
-
+console.log('Exercises 12.1:', string1Exercise12.endsWith('!'));
+console.log('Exercises 12.2:', string2Exercise12.endsWith('!'));
 
 console.log('------------------------');
 // 13. Use the `includes` method to check if a string includes a "%" character and print the result to the console.
 const string1Exercise13 = "$9.99 - Sale! 50% off all items in store";
 const string2Exercise13 = "Bundle deal: Save $50 when you buy two products together!";
+console.log('Exercises 13.1:', string1Exercise13.includes("%"));
+console.log('Exercises 13.2:', string2Exercise13.includes("%"));
 
 
 console.log('------------------------');
 // 14. Use the `substring` method to extract the word "Databases" from string stringExercise14 based on two indices and print it to the console.
 // Question: What is the difference between `substring()` and `slice()`?
 const stringExercise14 = 'Databases are an essential component of many applications.';
+console.log('Exercises 14:', stringExercise14.substring(9));
+
+
 
 console.log('------------------------');
 // 15. Remove extra spaces from the string stringExercise15 and print the result to the console.
 const stringExercise15 = "    Hello world    ";
+console.log('Exercises 15:', stringExercise15.trim());
+
+
 
 console.log('------------------------');
 // 16. Extract the price and currency from a string below using "slice" method and print it to console.
 const planPriceString = "Premium plan - 9.99 USD/month."; // Should return "9.99 USD"
+console.log('Exercises 16:', planPriceString.slice(-15,-7));
```

Comment:
This solution works but is very fragile. You'd usually search for a certain pattern in the string instead. 
Try something like this:

- Find the starting index of the price (e.g. search for '-', don't forget about the space right after it!)
- Find the ending index of the price (e.g. search for '/')
- Extract the price and currency substring (use slice)
- Print the extracted price and currency